clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/description/KubernetesCoordinates.java

```diff
@@ -70,4 +71,19 @@ public KubernetesCoordinatesBuilder fullResourceName(String fullResourceName) {
       return this;
     }
   }
+
+  /**
+   * Given a full KubernetesManifest object, parses out the kind, namespace, and name to create a
+   * corresponding KubernetesCoordinates object.
+   *
+   * @param manifest the manifest to parse
+   * @return the KubernetesCoordinates object
+   */
+  public static KubernetesCoordinates fromManifest(KubernetesManifest manifest) {
```

Comment:
💯 I kind of wished this function existed in one of my refactors last week, but didn't actually write it.